src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -166,6 +166,34 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    void TerminalPage::_HandleCloseOtherPanes(const IInspectable& /*sender*/,
+                                              const ActionEventArgs& args)
+    {
+        if (const auto terminalTab{ _GetFocusedTabImpl() })
+        {
+            const auto activePane = terminalTab->GetActivePane();
+            if (terminalTab->GetRootPane() != activePane)
+            {
+                _UnZoomIfNeeded();
+
+                // Accumulate list of all unfocused leaf panes
+                std::vector<uint32_t> unfocusedPaneIds;
+                terminalTab->GetRootPane()->WalkTree([&](auto&& p) {
+                    const auto id = p->Id();
+                    if (id.has_value() && id != activePane->Id())
+                    {
+                        unfocusedPaneIds.push_back(id.value());
+                    }
+                });
+
+                // Start by removing the panes that were least recently added
+                sort(begin(unfocusedPaneIds), end(unfocusedPaneIds), std::less<uint32_t>());
+                _CloseUnfocusedPanes(terminalTab->get_weak(), std::move(unfocusedPaneIds));
+            }
+        }
+        args.Handled(true);

```

Comment:
Changed this in f9ad9f2. The function now marks the arguments as handled if and only if the active pane is identical to the root pane (otherwise it is set to false)

Reply:
As of 73f02ef, this is now only the case for situations where there are panes that can be closed (non read-only unfocused panes)